### PR TITLE
Add 'strict_double_quotes' tunable to forbid use of double-quoted string literals

### PR DESCRIPTION
**Pull request by mistachkin**
These changes add a new tunable named 'strict_double_quotes'.  When this tunable is enabled, any attempt to use a double-quoted string literal will result in a SQL syntax error.

### CODE REVIEW

File: tests/strict_dbl_quotes.test/t02.req.out [hunk 1]
```diff
@@ -0,0 +1,21 @@
+('------- PHASE 1 -------'='------- PHASE 1 -------')
+("1"='1')
+("x"='x')
+(x='1')
+("y"='y')
+('------- PHASE 2 -------'='------- PHASE 2 -------')
+[SELECT "1"] failed with rc -3 double-quoted string literal: "1"
+[SELECT "x"] failed with rc -3 double-quoted string literal: "x"
+(x='1')
+[SELECT "y" FROM t1 WHERE x = 1] failed with rc -3 double-quoted string literal: "y"
```
Comment (nirbhayc) [inline tests/strict_dbl_quotes.test/t02.req.out]: This should have passed. We're only forbidding double-quoted string literals. Double-quoted identifiers are perfectly valid (and also SQL standard).

File: sqlite/src/resolve.c [hunk 2]
```diff
@@ -514,6 +515,12 @@ static int lookupName(
   if( cnt==0 && zTab==0 ){
     assert( pExpr->op==TK_ID );
     if( ExprHasProperty(pExpr,EP_DblQuoted) ){
+#if defined(SQLITE_BUILDING_FOR_COMDB2)
```
Comment (nirbhayc) [inline sqlite/src/resolve.c]: We should only be forbidding double-quoted TK_STRINGs (and not TK_IDs).

File: tests/strict_dbl_quotes.test/t02.req.out (hunk 1 above)
Comment (mistachkin) [reply tests/strict_dbl_quotes.test/t02.req.out]: Which query should have passed here?  The one on line 10?  For that one, the t1 table has no column "y".  Several others are similar.

File: sqlite/src/resolve.c (hunk 2 above)
Comment (mistachkin) [reply sqlite/src/resolve.c]: If I recall correctly, the token code here has more to do with the expectation than reality.  I think it's basically just the difference between single and double quotes, as far as the tokenizer is concerned.

File: tests/strict_dbl_quotes.test/t02.req.out (hunk 1 above)
Comment (mistachkin) [reply tests/strict_dbl_quotes.test/t02.req.out]: Just added some more tests to verify that valid double-quoted identifiers are working correctly with the tunable enabled.

Comment (nirbhayc) [reply tests/strict_dbl_quotes.test/t02.req.out]: I stand corrected. I thought "y" was a column in t1.

File: tests/strict_dbl_quotes.test/lrl.options [hunk 3]
```diff
@@ -0,0 +1 @@
+enable_sql_stmt_caching NONE
```
Comment (nirbhayc) [inline tests/strict_dbl_quotes.test/lrl.options]: Why?

File: sqlite/src/resolve.c (hunk 2 above)
Comment (nirbhayc) [reply sqlite/src/resolve.c]: Ok, while looking at this patch, I wrongly assumed that we haven't yet looked up for a matching column name at this point. I'm ok with this change.

File: tests/strict_dbl_quotes.test/lrl.options (hunk 3 above)
Comment (mistachkin) [reply tests/strict_dbl_quotes.test/lrl.options]: Without this, the previously prepared statements that had double-quoted string literals will not be re-prepared, allowing it to pass incorrectly.

Comment (nirbhayc) [reply tests/strict_dbl_quotes.test/lrl.options]: makes sense.

**Pull request merged**